builtin/game/register.lua

```diff
@@ -79,8 +79,22 @@ local function check_modname_prefix(name)
 	end
 end
 
+local function check_node_list(list, field)
+	local t = type(list)
+	if t == "table" then
+		for _, entry in pairs(list) do
+			assert(type(entry) == "string" or type(entry) == "number",
+				"Field '" .. field .. "' contains non-string entry")
+		end
+	elseif t ~= "string" and t ~= "number" and t ~= "nil" then
```

Comment:
Same question: When can it be a string or number? That's undocumented, hence should not be allowed.

Reply:
In this case numbers must be allowed for backwards compatibility. This code accepts either strings or numbers: https://github.com/minetest/minetest/blob/b89eb605b7020998fbf5d7142033ceb5275b3f71/src/script/cpp_api/s_env.cpp#L130-L131

Reply:
There's no need for backwards compatibility of undocumented behaviour.